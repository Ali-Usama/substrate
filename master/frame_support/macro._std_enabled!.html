<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro._std_enabled!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro._std_enabled!.html'">
    <meta http-equiv="refresh" content="0;URL=macro._std_enabled.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro._std_enabled.html">macro._std_enabled.html</a>...</p>
    <script>location.replace("macro._std_enabled.html" + location.search + location.hash);</script>
</body>
</html>