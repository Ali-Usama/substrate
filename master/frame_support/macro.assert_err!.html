<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_err!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_err!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.assert_err.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.assert_err.html">macro.assert_err.html</a>...</p>
    <script>location.replace("macro.assert_err.html" + location.search + location.hash);</script>
</body>
</html>