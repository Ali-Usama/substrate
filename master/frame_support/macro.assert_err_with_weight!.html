<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_err_with_weight!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_err_with_weight!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.assert_err_with_weight.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.assert_err_with_weight.html">macro.assert_err_with_weight.html</a>...</p>
    <script>location.replace("macro.assert_err_with_weight.html" + location.search + location.hash);</script>
</body>
</html>