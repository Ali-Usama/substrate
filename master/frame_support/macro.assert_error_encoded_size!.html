<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_error_encoded_size!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_error_encoded_size!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.assert_error_encoded_size.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.assert_error_encoded_size.html">macro.assert_error_encoded_size.html</a>...</p>
    <script>location.replace("macro.assert_error_encoded_size.html" + location.search + location.hash);</script>
</body>
</html>