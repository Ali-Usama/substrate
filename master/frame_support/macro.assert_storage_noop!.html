<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_storage_noop!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.assert_storage_noop!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.assert_storage_noop.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.assert_storage_noop.html">macro.assert_storage_noop.html</a>...</p>
    <script>location.replace("macro.assert_storage_noop.html" + location.search + location.hash);</script>
</body>
</html>