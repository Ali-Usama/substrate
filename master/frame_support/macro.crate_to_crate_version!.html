<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.crate_to_crate_version!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.crate_to_crate_version!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.crate_to_crate_version.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.crate_to_crate_version.html">macro.crate_to_crate_version.html</a>...</p>
    <script>location.replace("macro.crate_to_crate_version.html" + location.search + location.hash);</script>
</body>
</html>