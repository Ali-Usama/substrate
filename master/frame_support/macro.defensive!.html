<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.defensive!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.defensive!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.defensive.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.defensive.html">macro.defensive.html</a>...</p>
    <script>location.replace("macro.defensive.html" + location.search + location.hash);</script>
</body>
</html>