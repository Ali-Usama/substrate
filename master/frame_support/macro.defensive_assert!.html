<!DOCTYPE html>
<html lang="en">
<head><link rel="canonical" href="https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.defensive_assert!.html"><meta http-equiv="refresh" content="0;URL='https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.defensive_assert!.html'">
    <meta http-equiv="refresh" content="0;URL=macro.defensive_assert.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.defensive_assert.html">macro.defensive_assert.html</a>...</p>
    <script>location.replace("macro.defensive_assert.html" + location.search + location.hash);</script>
</body>
</html>