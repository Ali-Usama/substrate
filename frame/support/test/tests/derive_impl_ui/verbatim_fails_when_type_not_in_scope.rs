use frame_support::{*, pallet_macros::verbatim};
use static_assertions::assert_type_eq_all;

pub trait Shape {
    type Area;
}

struct DefaultShape;

#[register_default_impl(DefaultShape)]
impl Shape for DefaultShape {
    #[verbatim]
    type Area = ();
}

struct Circle;

#[derive_impl(DefaultShape)] // Injects type Area = Area;
impl Shape for Circle {}

assert_type_eq_all!(<Circle as Shape>::Area, ());

fn main() {}
