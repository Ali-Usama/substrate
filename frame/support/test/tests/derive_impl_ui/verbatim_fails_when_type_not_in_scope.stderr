error[E0412]: cannot find type `Area` in this scope
  --> tests/derive_impl_ui/verbatim_fails_when_type_not_in_scope.rs:13:10
   |
13 |     type Area = ();
   |          ^^^^ help: you might have meant to use the associated type: `Self::Area`
...
18 | #[derive_impl(DefaultShape)] // Injects type Area = Area;
   | ---------------------------- in this macro invocation
   |
   = note: this error originates in the macro `__export_tokens_tt_default_shape` which comes from the expansion of the macro `frame_support::macro_magic::forward_tokens` (in Nightly builds, run with -Z macro-backtrace for more info)
